src/components/modals/EntityCreatorEditor/EntityCreatorContainer.tsx

```diff
@@ -351,6 +345,8 @@ class Container extends React.Component<Props, ComponentState> {
     onGetNameErrorMessage = (value: string): string => {
         const { intl } = this.props
 
+        const isNameUnrestricted = this.state.entityTypeVal !== CLM.EntityType.LUIS && this.state.entityTypeVal !== CLM.EntityType.LOCAL && this.state.entityTypeVal !== CLM.EntityType.ENUM
```

Comment:
Why is this check needed? I thought we never wanted people to name an entity starting with `builtin-`.  Granted now that we don't have any logic based on the name, only the entityType, I think it should be Ok to do it but I wanted to know the reasoning?

If we must keep this check a few things to change:
1 - It would be more readable and editable to use a list and `includes`.  I think this would work:
```
const isNameUnrestricted = [CLM.EntityType.LUIS, CLM.EntityType.LOCAL, CLM.EntityType.ENUM].includes(this.state.entityTypeVal)
```  
Side node if you have this many boolean expressions it's better to put each one on it's own line like so:
```
const isNameUnrestricted = this.state.entityTypeVal !== CLM.EntityType.LUIS
    && this.state.entityTypeVal !== CLM.EntityType.LOCAL
    && this.state.entityTypeVal !== CLM.EntityType.ENUM
```
Again easier to read, edit, compare, etc...

2. This variable (`isNameUnrestricted`) should be located as closely to where it's used as possible.
In other words, don't declare it up here. Move it down to line `357`.  It's easier to people to see the whole .concept, otherwise they have to see a variable know it's not used and keep it in their head and then remember what it was later down the file.

Reply:
Oh I see, if the entity is a prebuilt then we must allow that prefix. Is that correct?

If so then maybe it's better to invert the logic and avoid listing of entity types which may change make the logic incorrect. We want to know when an entity is not a prebuilt and then apply the extra check.  Not assume if it's not those types it must be a prebuilt.  It's the same conversation we had the other day.  You want your logic to be precise and positive assertion when possible.

```
// Only apply builtin- check if entity is a prebuilt
if (typeof this.state.entityType !== 'string') {
    && value.toLowerCase().startsWith(EntityCreatorEditor.PrebuiltPrefix) {
  return Util.formatMessageId(intl, FM.ENTITYCREATOREDITOR_FIELDERROR_RESERVED)
}
```
Maybe we can use the helpers Lars added `isPrebuilt` although I'm not sure if we have the needed data/input type.

Reply:
Threading the middle ground needle between reviewer comments